.githooks/pre-commit

```
@@ -7,16 +7,48 @@ if ! git diff --quiet || ! git diff --quiet -- .; then
     git stash push -q --keep-index --include-untracked -m precommit-format
 fi
 
-readarray -d '' files < <(
+# readarray -d '' files < <(
+#     git diff --cached --name-only -z --diff-filter=ACM \
+#     | grep -zE '\.py$' \
+#     | grep -zEv '^(llvm-project/|external/llvm-project/)'
+# )
+
+# if (( ${#files[@]} )); then
+#     if command -v yapf >/dev/null 2>&1; then
+#         echo "Running yapf on ${#files[@]} files..."
+#         yapf -i "${files[@]}"
+#         echo "Staging changes..."
+#         git add -- "${files[@]}"
+#     else
+#         echo "yapf not found, skipping Python formatting."
+#     fi
+# fi
+
+readarray -d '' cxxfiles < <(
     git diff --cached --name-only -z --diff-filter=ACM \
-    | grep -zE '\.py$' \
+    | grep -zE '\.(c|cc|cxx|cpp|hpp|h|hh|hxx|hip|cl)$'
     | grep -zEv '^(llvm-project/|external/llvm-project/)'
 )
-if (( ${#files[@]} )); then
-    echo "Running yapf on ${#files[@]} files..."
-    yapf -i "${files[@]}"
-    echo "Staging changes..."
-    git add -- "${files[@]}"
+
+if (( ${#cxxfiles[@]} )); then
+    echo "Running clang-format on ${#cxxfiles[@]} staged C/C++ files..."
+    # Use direct clang-format to avoid re-diff logic; rely on available binary
+    clangfmt_bin="${CLANG_FORMAT_BIN:-clang-format}"
+    if ! command -v "$clangfmt_bin" >/dev/null 2>&1; then
+        echo "Warning: $clangfmt_bin not found, skipping C/C++ formatting."
```

can we fail instead of a warning?